S2VX.Game/Story/Note/EditorHoldNote.cs

```diff
@@ -29,6 +29,15 @@ public class EditorHoldNote : HoldNote {
             HitSoundTimes = new List<double>() { HitTime, EndTime };
         }
 
+        public override void UpdateEndTime(double endTime) {
+            if (endTime < HitTime) {
+                return;
+            }
+            EndTime = endTime;
+            ((HoldApproach)Approach).EndTime = EndTime;
```

Comment:
Can you see if you can do an override of the Approach property for hold notes? This way we don't have to cast it like this. I think you can set Approach to virtual inside of S2VXNote.cs and then override it in HoldNote.cs. For more details: https://stackoverflow.com/a/9238902/13183186

Reply:
Is this good?

Reply:
I think this is okay. Feels kind of odd having 2 properties point to the same thing. I was hoping that we'd be able to somehow elegantly override the same property as a HoldApproach, but don't think that's possible.

Reply:
Yeah, not possible to override the same property name with a different type.